packages/matter.js/src/storage/StringifyTools.ts

```diff
@@ -57,6 +57,9 @@ export function toJson(object: SupportedStorageTypes): string {
         if (value instanceof Uint8Array) {
             return `{"${JSON_SPECIAL_KEY_TYPE}":"Uint8Array","${JSON_SPECIAL_KEY_VALUE}":"${value.toHex()}"}`;
         }
+        if (value != null && typeof value === 'object' && value.type === 'Buffer' && value.data instanceof Array) {
```

Comment:
is that really already json-ified value in this place?

In any case please add a comment here why we do this because we would add logic for Buffers (which should not be part of matter.js) but needed here ...

Reply:
> is that really already json-ified value in this place?

So, this was the part that confused me a bit. Its not JSON per se , the `JSON.stringify` replacer function calls `toJson` if that function exists on a object its iterating on , so the `value` here is not an actual Buffer  but an object.  Whats more confusing is that `toJSON` does not actually return JSON, but rather a object that is intended to be JSON serialized, its a badly named function imho.  You can see that at https://github.com/nodejs/node/blob/c55aa01e4873152ea6a0dbff98d55d403b6b97bb/lib/buffer.js#L1135